gql/comparator_test.go

```diff
@@ -0,0 +1,279 @@
+package gql
+
+import "testing"
+
+func TestOperators(t *testing.T) {
+	tests := []struct {
+		description          string
+		operation            NewListOperation
+		arguments            map[string]interface{}
+		operand              interface{}
+		wantInvalidOperation bool
+		want                 bool
+	}{
+		{
+			description: "string equal valid and true",
+			operation:   NewEqualComparator,
+			arguments:   map[string]interface{}{"Value": "a"},
+			operand:     "a",
+			want:        true,
+		},
+		{
+			description: "string equal valid and false",
+			operation:   NewEqualComparator,
+			arguments:   map[string]interface{}{"Value": "a"},
+			operand:     "b",
+			want:        false,
+		},
+		{
+			description: "string equal invalid, mismatched operand",
+			operation:   NewEqualComparator,
+			arguments:   map[string]interface{}{"Value": "a"},
+			operand:     3,
+			want:        false,
+		},
+		{
+			description:          "NewEqualComparator invalid argument, nil",
+			operation:            NewEqualComparator,
+			arguments:            nil,
+			wantInvalidOperation: true,
```

Comment:
would "" be treated as an invalidOperation or just return a false?

Reply:
the arguments needs to be a `map[string]interface{}` so "" would be a compile error

Reply:
I meant something like `map[string]interface{}{"Value": ""},`

Reply:
Oh that would be valid, a comparison to the empty string. I'll add in a case for it.